src/ripple/rpc/handlers/GatewayBalances.cpp

```diff
@@ -78,6 +78,12 @@ doGatewayBalances(RPC::JsonContext& context)
 
     result[jss::account] = toBase58(accountID);
 
+    if (!ledger->exists(keylet::account(accountID)) && context.apiVersion > 1u)
```

Comment:
Just noticed here, apiVersion check should be the first check to avoid unnecessary checks.